src/api/uuid.rs

```diff
@@ -0,0 +1,121 @@
+use uuid::Uuid;
+
+const BLUETOOTH_BASE_UUID: u128 = 0x00000000_0000_1000_8000_00805f9b34fb;
+const BLUETOOTH_BASE_MASK: u128 = 0x00000000_ffff_ffff_ffff_ffffffffffff;
+const BLUETOOTH_BASE_MASK_16: u128 = 0xffff0000_ffff_ffff_ffff_ffffffffffff;
+
+/// Convert a 32-bit BLE short UUID to a full 128-bit UUID by filling in the standard Bluetooth Base
+/// UUID.
+pub const fn uuid_from_u32(short: u32) -> Uuid {
+    Uuid::from_u128(BLUETOOTH_BASE_UUID | ((short as u128) << 96))
+}
+
+/// Convert a 16-bit BLE short UUID to a full 128-bit UUID by filling in the standard Bluetooth Base
+/// UUID.
+pub const fn uuid_from_u16(short: u16) -> Uuid {
+    uuid_from_u32(short as u32)
+}
+
+/// An extension trait for `Uuid` which provides BLE-specific methods.
+pub trait BleUuid {
+    /// If the UUID is a valid BLE short UUID then return its short form, otherwise return `None`.
+    fn to_ble_u32(&self) -> Option<u32>;
+
+    /// If the UUID is a valid 16-bit BLE short UUID then return its short form, otherwise return
+    /// `None`.
+    fn to_ble_u16(&self) -> Option<u16>;
+
+    /// Convert the UUID to a string, using short format if applicable.
+    fn succinctly(&self) -> String;
+}
+
+impl BleUuid for Uuid {
+    fn to_ble_u32(&self) -> Option<u32> {
+        let value = self.as_u128();
+        if value & BLUETOOTH_BASE_MASK == BLUETOOTH_BASE_UUID {
+            Some((value >> 96) as u32)
+        } else {
+            None
+        }
+    }
+
+    fn to_ble_u16(&self) -> Option<u16> {
+        let value = self.as_u128();
+        if value & BLUETOOTH_BASE_MASK_16 == BLUETOOTH_BASE_UUID {
+            Some((value >> 96) as u16)
+        } else {
+            None
+        }
+    }
+
+    fn succinctly(&self) -> String {
+        if let Some(uuid16) = self.to_ble_u16() {
+            format!("{:#04x}", uuid16)
```

Comment:
Since the printed number may not be obviously hexadecimal, would prefixing them with `0x` be appropriate? A supporting argument is, in the official Bluetooth Assigned Numbers document, all the 16bit UUIDs are also prefixed with `0x`

Reply:
Forgot to suggest adding this change for 32bit UUIDs as well

Reply:
This does prefix them with `0x`.